packages/codec/lib/format/utils/inspect.ts

```diff
@@ -319,9 +363,27 @@ export function nativizeVariables(variables: {
   );
 }
 
-//HACK! Avoid using! Only use this if:
-//1. you absolutely have to, or
-//2. it's just testing, not real code
+//HACK! Avoid using!
+/**
+ * WARNING! Do NOT use this function in real code unless you absolutely have
+ * to!  Using it in controlled tests is fine, but do NOT use it in real code if
+ * you have any better option!
+ *
+ * This function is a giant hack.  It will throw exceptions on numbers that
+ * don't fit in a Javascript number.  It will go into an infinite loop on
+ * circular structures (although that might be fixed eventually).  It was only
+ * ever written to support our hacked-together watch expression system, and
+ * later repurposed to make testing easier.
+ *
+ * If you are not doing something as horrible as our hacked-together watch
+ * expression system, then you probably have a better option than using this in
+ * real code!
+ *
+ * Remember, the decoder output format was made to be machine-readable.  It
+ * shouldn't be too hard for you to process.  If it comes to it, copy-paste
+ * this code and dehackify it for your use case, which hopefully is saner than
+ * the one that caused us to write this.
+ */
```

Comment:
This should probably end with "if you want to just look at these values, see ResultsInspector"